### PR TITLE
Create new dedicated Rollup Action Group Config

### PR DESCRIPTION
**Pull request by talevy**
This commit migrates RollupAction away from using the
legacy Rollup's GroupConfig and DateHistogramGroupConfig
classes. RollupAction is not concerned with legacy intervals
and delays.

### CODE REVIEW

**Review by csoulios — APPROVED** [review]
LGTM!

Left minor comments

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/rollup/RollupActionDateHistogramGroupConfig.java [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+ *     ]
+ * }
+ */
+public abstract class RollupActionDateHistogramGroupConfig implements Writeable, ToXContentObject {
+
+    static final String NAME = "date_histogram";
+    public static final String FIXED_INTERVAL = "fixed_interval";
+    public static final String CALENDAR_INTERVAL = "calendar_interval";
+    public static final String TIME_ZONE = "time_zone";
+    public static final String DELAY = "delay";
```
[the remark this answers is not in the document]
Comment (talevy) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/rollup/RollupActionDateHistogramGroupConfig.java:55]: good catch! the usage of this was removed, but I seem to have missed removing this

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/rollup/RollupTask.java [hunk 2]
```diff
@@ -21,7 +20,7 @@
     private RollupActionConfig config;
     private RollupJobStatus status;
 
-    RollupTask(long id, String type, String action, TaskId parentTask, String rollupIndex, RollupActionConfig config,
+    public RollupTask(long id, String type, String action, TaskId parentTask, String rollupIndex, RollupActionConfig config,
```
[the remark this answers is not in the document]
Comment (talevy) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/rollup/RollupTask.java]: 👍 moved it into the same package as the RollupAction. The new rollup-implementation will remove these anyways :)

**Review by csoulios — APPROVED** [review]
LGTM! 
Left two very minor comments. 

Thanks Tal!

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/rollup/RollupActionDateHistogramGroupConfig.java [hunk 3]
```diff
@@ -0,0 +1,308 @@
...
+
+/**
+ * The configuration object for the histograms in the rollup config
+ *
+ * {
+ *     "groups": [
+ *        "date_histogram": {
+ *            "field" : "foo",
+ *            "interval" : "1d",
+ *            "delay": "30d",
```
[the remark this answers is not in the document]
Comment (jrodewig) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/rollup/RollupActionDateHistogramGroupConfig.java:43]: `interval` should be `calendar_interval` too.

**Pull request merged**